components/suggestion/command_provider/applet_command_parser.ts

```diff
@@ -0,0 +1,589 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {getPluginsLocations} from 'mattermost-redux/selectors/entities/plugins';
+import {default as PluginLocationConst} from 'mattermost-redux/constants/plugins';
+import {GlobalState} from 'mattermost-redux/types/store';
+import {PluginLocation} from 'mattermost-redux/types/plugins';
+
+import * as Utils from 'utils/utils.jsx';
+
+import {
+    AppletCall,
+    AutocompleteDynamicSelect,
+    AutocompleteElement,
+    AutocompleteStaticSelect,
+    AppletBinding,
+    AppletContext,
+    AppletFormMetadataResponse,
+    AppletField,
+    AppletSelectOption,
+    AppletFieldTypes,
+} from 'actions/applet_types';
+
+import {EXECUTE_CURRENT_COMMAND_ITEM_ID} from './command_provider';
+import {doAppletCall} from 'actions/applets';
+
+export type SuggestionChoice = {
+    suggestion: string;
+    complete?: string;
+    getComplete?: (pretext: string, value: string) => string;
+    description?: string;
+    hint?: string;
+    iconData?: string;
+}
+
+export const decorateSuggestionChoiceComplete = (pretext: string, choice: SuggestionChoice): SuggestionChoice => {
+    if (choice.complete?.endsWith(EXECUTE_CURRENT_COMMAND_ITEM_ID)) {
+        return choice;
+    }
+
+    const words = pretext.split(' ');
+    words.pop();
+    const before = words.join(' ') + ' ';
+
+    if (choice.getComplete) {
+        choice.complete = choice.getComplete(pretext, choice.suggestion);
+    }
+    choice.hint = choice.hint || '';
+
+    if (!words.length) {
+        // base command
+        return choice;
+    }
+
+    return {
+        ...choice,
+        suggestion: '/' + choice.suggestion,
+        complete: before + choice.suggestion,
+    };
+}
+
+const extractBaseCommand = (pretext: string) => {
+    return pretext.substring(1).split(' ')[0];
+}
+
+export const shouldHandleCommand = (pretext: string, bindings: AppletBinding[]): boolean => {
+    for (const binding of bindings) {
+        let trigger = binding.name;
+        if (!trigger) {
+            continue;
+        }
+
+        if (trigger[0] !== '/') {
+            trigger = '/' + trigger;
+        }
+
+        if (pretext.startsWith(trigger + ' ')) {
```

Comment:
is `pretext` just the part of the current word, or the entire command line prior? Suggest a more intention-revealing name, depending on the semantics.

Also 1/5 suggest s/trigger/name/ for consistency, there is no need for another term.

Reply:
`pretext` refers to all text to the left of the user's cursor. It is used throughout the codebase as such already.

Trigger point addressed